### ISSUE TITLE
ML-agents python low-level API

### ISSUE THREAD

**Issue by X-DDDDD** [Unity-Technologies/ml-agents#5564]
Hello, ML agents developer! I'm very happy to use your training tools,but I have some problems in using them at present.

I'm trying to use [Python-API.md](https://github.com/Unity-Technologies/ml-agents/blob/release_12/docs/Python-API.md) to test my RL algorithm,but when I build the example containing 12 scenes into an executable file according to [Learning-Environment-Executable.md](https://github.com/Unity-Technologies/ml-agents/blob/release_12/docs/Learning-Environment-Executable.md), I encounter difficulties in getting the state input in the algorithm. [Terminalsteps](https://github.com/Unity-Technologies/ml-agents/blob/release_12/docs/Python-API.md#terminalsteps-and-terminalstep) cannot return all the observation, reward and other information of the 12 scenes. It will only return information about a scene(In 12 scenes, other senses that are not triggered will also be reset) that goes to Terminalsteps.

In other words,according to the interface provided by [Python-API.md](https://github.com/Unity-Technologies/ml-agents/blob/release_12/docs/Python-API.md), when the ball of one scene falls, all 12 scenes will be reset, so I can't carry out parallel training(Even if the ball of one scene falls, other scenes are still in progress as in unity).

So if I want to use my own algorithm to train multiple scenes(an agent) in the [Example Learning Environments](https://github.com/Unity-Technologies/ml-agents/blob/release_12/docs/Learning-Environment-Examples.md), how should I use your tools?

Thanks! :)

**Comment by andrewcoh** [comment]
Hi @X-DDDDD 

Thank you for using our Python API. The TerminalSteps contains _only_ information about agents that have reached a termination condition (i.e. EndEpisode() called in C#).  The information for the parallel agents that have not reached a termination condition will be contained in the DecisionSteps object.

Are you seeing behavior that is different from this? Please share as much detail as you can.

**Comment by X-DDDDD** [comment]
> Hey guys, can you provide some advice on this, I cannot use this API with multiple scenes either?

A lot of people have this problem now.🤣

**Comment by X-DDDDD** [comment]
@andrewcoh 
很抱歉再次打扰你

**Comment by X-DDDDD** [comment]
@andrewcoh 

I'm sorry to bother you again.

I can't make input under continuous actions now. How should I modify this?
```bash
action = ActionTuple.add_continuous(continuous=(np.array([0.1,0.1,0.3], dtype=float)))
env.set_actions(behavior_name, action)
```

Thanks!

**Comment by MarcoMeter** [comment]
HI @X-DDDDD 
the most reliable way to test your custom algorithm is only to use one agent inside one build instance. Being in need of multiple agents/environments, you have to run multiple build instances concurrently.

The Python API is not very intuitive when it comes to handling multiple agents. The sum of DecisionSteps and TerminalSteps is not guaranteed to equal your total agent count, even though if all agents share the same behavior.

@andrewcoh 
Correct me if I'm wrong.
